File: app/src/main/java/com/silambar/parcelabledemo/java/TempleInfo.java
package com.silambar.parcelabledemo.java;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by silambarasan on 13/03/18.
 */

public class TempleInfo implements Parcelable {

  private String address;
  private String description;

  public TempleInfo(){}

  public String getAddress() {
    return address;
  }

  public void setAddress(String address) {
    this.address = address;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  protected TempleInfo(Parcel in) {
    address = in.readString();
    description = in.readString();
  }

  @Override public void writeToParcel(Parcel dest, int flags) {
    dest.writeString(address);
    dest.writeString(description);
  }

  @Override public int describeContents() {
    return 0;
  }

  public static final Creator<TempleInfo> CREATOR = new Creator<TempleInfo>() {
    @Override public TempleInfo createFromParcel(Parcel in) {
      return new TempleInfo(in);
    }

    @Override public TempleInfo[] newArray(int size) {
      return new TempleInfo[size];
    }
  };
}

File: app/src/main/java/com/silambar/parcelabledemo/kotlin/TempleInfo.kt
package com.silambar.parcelabledemo.kotlin

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator

/**
 * Created by silambarasan on 13/03/18.
 */

class TempleInfo : Parcelable {

  var address: String? = null
  var description: String? = null

  constructor() {}

  protected constructor(parcel: Parcel) {
    address = parcel.readString()
    description = parcel.readString()
  }

  override fun writeToParcel(dest: Parcel, flags: Int) {
    dest.writeString(address)
    dest.writeString(description)
  }

  override fun describeContents(): Int {
    return 0
  }

  companion object CREATOR : Creator<TempleInfo> {
    override fun createFromParcel(parcel: Parcel): TempleInfo {
      return TempleInfo(parcel)
    }

    override fun newArray(size: Int): Array<TempleInfo?> {
      return arrayOfNulls(size)
    }
  }
}
